libs/wingsdk/src/cloud/on-deploy.md

```diff
@@ -0,0 +1,52 @@
+---
+title: OnDeploy
+id: on-deploy
+description: A resource that runs inflight code during the application's deployment.
+keywords:
+  [
+    Wing reference,
+    Wing language,
+    language,
+    Wing standard library,
+    Wing programming language,
+    OnDeploy,
+    Trigger,
+    Deployment,
+  ]
+sidebar_position: 1
+---
+
+The `cloud.OnDeploy` resource runs a block of inflight code each time the application is deployed.
+
+## Usage
+
+```ts playground
+bring cloud;
+
+let bucket = new cloud.Bucket();
+
+// each time the application is deployed, all objects in the bucket are deleted
+let setup = new cloud.OnDeploy(inflight () => {
```

Comment:
I don't like `OnDeploy` but I really don't have a better name, so let's go with it.